### DISCUSSION TITLE
Can't set environment variable from secret

### DISCUSSION THREAD

**Discussion by g105b** [community/community#26339] [Actions]
<p>Hello all,</p>
<p>I’m trying to pass in a secret value to my script that’s running using a composite run step, but I’m getting the following error message:</p>
<p><code>"Unrecognized named-value: 'secrets'.</code></p>
<p>Here’s the yaml file: <a href="https://github.com/php-actions/composer/blob/434c409f860b690aa093760b78cdb252e78dd5d9/action.yml#L95-L96" rel="noopener nofollow ugc">https://github.com/php-actions/composer/blob/434c409f860b690aa093760b78cdb252e78dd5d9/action.yml#L95-L96</a></p>
<p>As you can see, I’m successfully passing a few environment variables into the script from the value of the Action’s input parameters, but I can’t figure out how to set the environment variables from secrets.</p>
<p>Am I doing something wrong? Is there a bug with Github actions?</p>
<p>Any help will be appreciated, thanks!</p>
<p>Here’s an example repository running this action: <a href="https://github.com/php-actions/example-phpunit/runs/1485735187" rel="noopener nofollow ugc">https://github.com/php-actions/example-phpunit/runs/1485735187</a></p>

**Comment by otter-computer** [comment, marked as answer]
<p>It looks like you’re trying to use secrets from inside an action definition file rather than inside a workflow. Secrets aren’t available directly inside a custom action, they need to be passed into a custom action input from a workflow file.</p>
<p>You should be able to define these as an input inside your <code>action.yml</code> and then pass your required secrets to this action from inside the workflow file you’re using this action in.</p>
